pkg/provider/azure_controller_vmssflex.go

```diff
@@ -38,7 +38,7 @@ func (fs *FlexScaleSet) AttachDisk(ctx context.Context, nodeName types.NodeName,
 		return nil, err
 	}
 
-	nodeResourceGroup, err := fs.GetNodeResourceGroup(vmName)
+	nodeResourceGroup, err := fs.GetNodeResourceGroup(*vm.Name)
```

Comment:
CSI Driver is  using nodeName as the unique key to identify nodes/VMs. Changing to vm.name might cause issue here if external resource group is used. Can you change back to use vmName?